docs/topics/spider-middleware.rst

```diff
@@ -157,6 +160,15 @@ object gives you access, for example, to the :ref:`settings <topics-settings>`.
         It receives an iterable (in the ``start_requests`` parameter) and must
         return another iterable of :class:`~scrapy.http.Request` objects.
 
+        When using Python 3.6+ `start_requests` may be an
+        :term:`python:asynchronous iterable`. To support such spiders, this
+        middleware method should be an :term:`python:asynchronous generator`,
+        which should support both synchronous and asynchronous
+        ``start_requests``. :func:`scrapy.utils.asyncgen.as_async_generator`
+        can be used to convert both kinds of ``start_requests`` to an
+        asynchronous iterable. See :ref:`more details here
+        <async-start_requests>`.
```

Comment:
If we use `autofunction` as I mention in an earlier comment, the “See …” sentence here could be removed, since `` :func:`scrapy.utils.asyncgen.as_async_generator` `` already links to the relevant part of the documentation.